app/ota_client.py

```diff
@@ -664,6 +717,52 @@ def _copy_persistent_files(self, list_file, standby_path):
             ):  # NOTE: not equivalent to perinf.path.exists()
                 copy_tree.copy_with_parents(perinf.path, standby_path)
 
+    def enter_update(self, version):
+        logger.debug("check if ota_status is valid for updating...")
+        self.check_update_status()
+
+        logger.debug("pre-update setup...")
+        self.boot_ctrl_pre_update(version)
+        self.set_ota_status(OtaStatus.UPDATING)
+        self.write_standby_ota_status(OtaStatus.UPDATING)
+        logger.debug("finished pre-update setup")
+
+    def leave_update(self):
+        logger.debug("post-update setup...")
+        self.boot_ctrl_post_update()
+
+    def enter_rollbacking(self):
+        self.check_rollback_status()
+        self.set_ota_status(OtaStatus.ROLLBACKING)
+        self.write_standby_ota_status(OtaStatus.ROLLBACKING)
+        self.boot_ctrl_pre_rollback()
+
+    def leave_rollbacking(self):
+        self.boot_ctrl_post_rollback()
+
+
+def ota_client_instance():
```

Comment:
This function doesn't return instance. returns class. So the function name should be ota_client_class.